File: LeetCode/287_FindTheDuplicateNumber.py
# Q) 287. Find the Duplicate Number.
# https://leetcode.com/problems/find-the-duplicate-number/description/

#code
class Solution:
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = set()
        for i in range(len(nums)):
            if nums[i] in l:
                return nums[i]
            else:
                l.add(nums[i])
        

File: LeetCode/287-Find the Duplicate Number.java
// 			287. Find the Duplicate Number
// 			https://leetcode.com/problems/find-the-duplicate-number/description/
//code
class Solution {
    public int findDuplicate(int[] nums) {
        
        Set<Integer> s = new HashSet<>();
        for(int i=0; i< nums.length; i++){

            if(s.contains(nums[i])){
                return nums[i];
            }else{
                s.add(nums[i]);
            }
        }
        
        return -1;
        
    }
}